### PR TITLE
Add lint script, workflow, and pre-commit hook

### PR DESCRIPTION
**Pull request by ohltyler**
### Description

Follow-up to #3 

- adds a linting workflow to run lint on all src files on PR and push
- adds `precommit` npm package and configuration to automatically run the linter on every commit

### Issues Resolved

[List any issues this PR will resolve]

### Check List

- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
- [ ] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+name: Linting workflow
+on:
+  push:
+    branches:
```
Comment (owaiskazi19) [inline .github/workflows/lint.yml:4]: I don't think so we need to specify `branches` when you want to run the workflow on all the branches anyway. I would still suggest to test the workflow on your fork and remove this line and 7th after that.

Comment (ohltyler) [reply .github/workflows/lint.yml]: Thanks for the suggestion. Updated and it is running as expected still.

File: public/app.tsx [hunk 2]
```diff
@@ -12,6 +12,7 @@ import { Navigation, APP_PATH } from './utils';
 import { Overview, UseCases, Workflows } from './pages';
 import { CoreServicesConsumer } from './core_services';
 
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
```
Comment (owaiskazi19) [inline public/app.tsx:15]: For my understanding: Is this comment needed to disable eslint? If so, why are disabling it here?

Comment (ohltyler) [reply public/app.tsx:15]: Its just to disable the next line which has an empty interface. Regularly the linting would flag this as an issue, but in this case it's expected.

common practice used in OSD which I'm following here is all

**Review by owaiskazi19 — APPROVED** [review]

**Pull request merged**